docs/reference/commandline/pull.md

```diff
@@ -27,6 +27,9 @@ can `pull` and try without needing to define and configure your own.
 To download a particular image, or set of images (i.e., a repository),
 use `docker pull`.
 
+> **Note**: During http configuration, engine uses the proxy indicated by 
+> $HTTP\_PROXY and $NO\_PROXY environment variables. See [document for http package](https://golang.org/pkg/net/http/#ProxyFromEnvironment) for details.
+
```

Comment:
Perhaps we should refer to https://github.com/docker/docker/blob/master/docs/admin/systemd.md#http-proxy.
Also we could make it a small section, instead of a note; something like;

```Markdown
## Proxy configuration

If you are behind a HTTP proxy server, for example in corporate settings, you
may need to configure the Docker daemon's proxy settings, using the `HTTP_PROXY`,
`HTTPS_PROXY`, and `NO_PROXY` environment variables. To set these environment
variables on a host using `systemd`, refer to the [control and configure Docker
with systemd](../../admin/systemd.md#http-proxy) documentation.
```